app/scripts/modules/core/src/pipeline/config/graph/PipelineGraph.tsx

```diff
@@ -373,6 +381,14 @@ export class PipelineGraph extends React.Component<IPipelineGraphProps, IPipelin
 
   public componentDidMount() {
     window.addEventListener('resize', this.windowResize);
+    this.mounted = true;
+    if (!this.state.hydrated && this.props.execution) {
+      ReactInjector.executionService.hydrate(this.props.application, this.props.execution).then(() => {
+        if (this.mounted) {
+          this.setState({ hydrated: true });
+        }
+      });
+    }
```

Comment:
one straightforward, relatively low-effort way to make use of Rx for this type of logic is to setup a `destroy$` observable. That gets you a slightly more readable control flow and avoids doing boolean flagging (the flagging doesn't seem complex enough to be error prone here but it can easily turn error prone with additional logic):

```typescript
import { Subject, Observable } from 'rxjs';

// ... class setup
private destroy$ = new Subject();

// ... in lifecycle methods
componentDidMount() {
  const { application, execution } = this.props;
  const { executionService } = ReactInjector;

  if (!this.state.hydrated && execution) {
    Observable.fromPromise(
      executionService.hydrate(application, execution)
    ).takeUntil(this.destroy$).subscribe(() => {
      this.setState({ hydrated: true });
    });
  }
}

componentWillUnmount() {
  this.destroy$.next();
}
```

Reply:
(if you choose, you can use this pattern wherever you're currently using a `mounted` flag)